tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/cluster/distribution/ClusterNotificationsTest.java

```diff
@@ -0,0 +1,124 @@
+   }
+
+   @Test
+   public void testClusterNotificationsContinuity() throws Exception {
+      final MessageLoadBalancingType lbType = MessageLoadBalancingType.OFF_WITH_REDISTRIBUTION;
+      final String namePrefix = "TEST.QUEUE.";
+      final int queueCount = 500;
+
+      AssertionLoggerHandler loggerHandler = new AssertionLoggerHandler();
+      runAfter(loggerHandler::close);
+
+      setupServer(0, isFileStorage(), isNetty());
+      setupServer(1, isFileStorage(), isNetty());
+      setupServer(2, isFileStorage(), isNetty());
+
+      setupClusterConnection("cluster0", "", lbType, 1, isNetty(), 0, 1, 2);
+      setupClusterConnection("cluster0", "", lbType, 1, isNetty(), 1, 2, 0);
+      setupClusterConnection("cluster0", "", lbType, 1, isNetty(), 2, 0, 1);
+
+      startServers(0, 1, 2);
+
+      AddressSettings as = new AddressSettings().setRedistributionDelay(0);
+      servers[0].getAddressSettingsRepository().addMatch(namePrefix + "#", as);
+      servers[1].getAddressSettingsRepository().addMatch(namePrefix + "#", as);
+      servers[2].getAddressSettingsRepository().addMatch(namePrefix + "#", as);
+
+      waitForTopology(servers[2], 3);
+
+      ExecutorService executorService = Executors.newCachedThreadPool();
+      CountDownLatch latch = new CountDownLatch(queueCount * 2);
+
+      try (ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory("tcp://localhost:61616")) {
+
+         List<DefaultMessageListenerContainer> containers = new ArrayList<>();
+         runAfter(() -> containers.parallelStream().forEach(AbstractJmsListeningContainer::shutdown));
+         runAfter(() -> executorService.shutdown());
+
+         for (int i = 0; i < queueCount; i++) {
+            Queue queue = ActiveMQDestination.createQueue(namePrefix + i + "-");
+
+            executorService.submit(() -> {
+               try (Connection connection = factory.createConnection();
+                    Session session = connection.createSession(Session.SESSION_TRANSACTED)) {
+
+                  session.createProducer(queue).send(session.createTextMessage("Message"));
+                  session.commit();
+                  latch.countDown();
+               } catch (Exception e) { }
+            });
+
+            DefaultMessageListenerContainer container = new DefaultMessageListenerContainer();
+            container.setCacheLevelName("CACHE_NONE");
+            container.setSessionTransacted(true);
+            container.setSessionAcknowledgeModeName("SESSION_TRANSACTED");
+            container.setConcurrentConsumers(20);
+            container.setConnectionFactory(new CachingConnectionFactory(factory));
+            container.setDestinationName(queue.getQueueName());
+            container.setBackOff(new FixedBackOff(1000, 2));
+            container.setReceiveTimeout(100);
+            container.setMessageListener((MessageListener) msg -> {
+               latch.countDown();
+            });
+
+            container.initialize();
```

Comment:
This test is failing on my laptop. not being able to create native threads...

Reply:
The problem here is (I guess) that in order for the test to trigger the issue this PR is about, underlying hardware most likely has to be working close to capacity. (This is not the case with an "actual" cluster though for some reason).
If you are interested in just trying this out I'd advice lowering queue count and consumer count on the Spring DMLC (while I see if I can find some formula to scale this based on available system resources or something).

Change would be:
:52   final int queueCount = 500; --> 100?

:100 container.setConcurrentConsumers(20); --> 5?